state-chain/pallets/cf-broadcast/src/lib.rs

```diff
@@ -102,21 +102,33 @@ pub mod pallet {
 	/// Type alias for the instance's configured ApiCall.
 	pub type ApiCallFor<T, I> = <T as Config<I>>::ApiCall;
 
-	#[derive(Clone, RuntimeDebug, PartialEq, Eq, Encode, Decode, TypeInfo)]
+	/// All data contained in a Broadcast
+	#[derive(RuntimeDebug, PartialEq, Eq, Encode, Decode, TypeInfo)]
 	#[scale_info(skip_type_params(T, I))]
-	pub struct BroadcastAttempt<T: Config<I>, I: 'static> {
+	pub struct BroadcastData<T: Config<I>, I: 'static> {
 		pub broadcast_id: BroadcastId,
 		pub transaction_payload: TransactionFor<T, I>,
 		pub threshold_signature_payload: PayloadFor<T, I>,
 		pub transaction_out_id: TransactionOutIdFor<T, I>,
 	}
 
-	/// The first step in the process - a transaction signing attempt.
+	impl<T: Config<I>, I: 'static> Clone for BroadcastData<T, I> {
+		fn clone(&self) -> Self {
+			BroadcastData {
+				broadcast_id: self.broadcast_id,
+				transaction_payload: self.transaction_payload.clone(),
+				threshold_signature_payload: self.threshold_signature_payload.clone(),
+				transaction_out_id: self.transaction_out_id.clone(),
+			}
+		}
+	}
+
+	/// Broadcast data with a nominated broadcaster
 	#[derive(Clone, RuntimeDebug, PartialEq, Eq, Encode, Decode, TypeInfo)]
 	#[scale_info(skip_type_params(T, I))]
-	pub struct TransactionSigningAttempt<T: Config<I>, I: 'static> {
-		pub broadcast_attempt: BroadcastAttempt<T, I>,
-		pub nominee: T::ValidatorId,
+	pub struct BroadcastWithNominee<T: Config<I>, I: 'static> {
```

Comment:
Does this really need to be a separate struct? Since nominee is optional anyway, can we not just flatten it into BroadcastData?

Reply:
Yeah Unifying these 2 structs makes everything simpler!